code/game/turfs/open/water.dm

```diff
@@ -138,8 +144,7 @@
 		return
 	enter_hot_spring(arrived)
 
-/turf/open/water/hot_spring/proc/enter_initialized_movable(datum/source, atom/movable/movable)
-	SIGNAL_HANDLER
+/turf/open/water/hot_spring/on_atom_inited(datum/source, atom/movable/movable)
 	if(!immerse_added && !is_type_in_typecache(movable, GLOB.immerse_ignored_movable))
 		AddElement(/datum/element/immerse, icon, icon_state, "immerse", immerse_overlay_color, alpha = immerse_overlay_alpha)
 		immerse_added = TRUE
```

Comment:
I'm pretty sure this will never run due to the code you added in Initialize, no?